arch/xtensa/src/esp32/esp32_wireless.c

```diff
@@ -288,6 +370,476 @@ void esp32_phy_disable_clock(void)
   leave_critical_section(flags);
 }
 
+#ifdef CONFIG_ESP32_SUPPORT_MULTIPLE_PHY_INIT_DATA
+
+/****************************************************************************
+ * Name: phy_crc_check
+ *
+ * Description:
+ *   Check the checksum value of data
+ *
+ * Input Parameters:
+ *   data     - Data buffer pointer
+ *   length   - Data length
+ *   checksum - Checksum pointer
+ *
+ * Returned Value:
+ *   OK on success; a negated errno on failure
+ *
+ ****************************************************************************/
+
+static int phy_crc_check(uint8_t *data, const uint8_t *checksum,
+                         size_t length)
+{
+  uint32_t crc_data = 0;
+  crc_data = crc32_le(crc_data, data, length);
+  uint32_t crc_size_conversion = htonl(crc_data);
+  if (crc_size_conversion != *(uint32_t *)(checksum))
```

Comment:
Can we change `checksum` to be `uint32_t *` to avoid cast?

Reply:
No, the uint8_t type pointer checksum is read from flash, if conversion is not done here, then conversion will need to be done on the checksum.

Reply:
I'm not sure if ESP32 supports unaligned access. If it is, then my comment can be ignored, but I see that
```
typedef struct {
    uint8_t control_info_checksum[4];     /*!< 4-byte control infomation checksum */
    uint8_t multiple_bin_checksum[4];     /*!< 4-byte multiple bin checksum */
    uint8_t check_algorithm;              /*!< check algorithm */
    uint8_t version;                      /*!< PHY init data bin version */
    uint8_t number;                       /*!< PHY init data bin number */
    uint8_t length[2];                    /*!< Length of each PHY init data bin */
    uint8_t reserved[19];                 /*!< 19-byte reserved  */
} __attribute__ ((packed)) phy_control_info_data_t;
```
So if unaligned access is not supported casting and dereferencing `checksum` to `uint_32_t *` may lead to unaligned access.
If unaligned access is not supported then we need to go with
```
uint32_t tmp = checksum[0] | (checksum[1] << 8) | (checksum[2] << 16) | (checksum[3] << 24);
if (crc_size_conversion != tmp)
```
to be safe